lib/chef/provider/aws_ebs_volume.rb

```diff
@@ -29,12 +39,67 @@ class Chef::Provider::AwsEbsVolume < Chef::Provider::AwsProvider
     if existing_volume
       converge_by "Deleting EBS volume #{fqn} in #{new_resource.region_name}" do
         existing_volume.delete
+
+        Retryable.retryable(:tries => 30, :sleep => 2, :on => TimeoutError) do
+          raise TimeoutError,
+            "Timed out waiting for volume #{new_resource.volume_id} to be deleted." if existing_volume.exists?
+        end
       end
     end
 
     new_resource.delete
   end
 
+  action :attach do
```

Comment:
@jkeiser, Cool. I see wanting to attach to an instance upon creation. I also see your point about `machine`.
How about:
```ruby
aws_ebs_volume 'volume' do
  machine 'blah'
end

aws_ebs_volume 'another_volume' do
  instance 'i-12345678'
end

```